# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLVideoElement.cpp
# Commit: e88d5d13ebef
# Full Hash: e88d5d13ebef251e0263a50a14b5bbdc58fae4a9
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-18 21:13:46
# Description:
#   Bug 1919423 - Check for null selected track when fulfilling rVFC. r=media-playback-reviewers,padenot
#   
#   HTMLVideoElement's mSelectedVideoStreamTrack may be null when fulfilling
#   requestVideoFrameCallback requests. This can occur when we race tearing
#   down the current stream, while there are still frames in the
# ==============================================================================

diff -r 3aed8cf9ce91 -r e88d5d13ebef dom/html/HTMLVideoElement.cpp
--- a/dom/html/HTMLVideoElement.cpp	Wed Sep 18 13:07:47 2024 +0000
+++ b/dom/html/HTMLVideoElement.cpp	Wed Sep 18 13:08:18 2024 +0000
@@ -798,7 +798,7 @@
   // timestamps relative to an origin that is specific to the WebRTC session.
   bool hasCaptureTimeNtp = selected->mWebrtcCaptureTime.is<int64_t>();
   bool hasReceiveTimeReal = selected->mWebrtcReceiveTime.isSome();
-  if (hasCaptureTimeNtp || hasReceiveTimeReal) {
+  if (mSelectedVideoStreamTrack && (hasCaptureTimeNtp || hasReceiveTimeReal)) {
     if (const auto* timestampMaker =
             mSelectedVideoStreamTrack->GetTimestampMaker()) {
       if (hasCaptureTimeNtp) {
